From c530bc1d555f4a16ad4dd10965258829a9f95b1e Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Mon, 28 Sep 2026 20:19:15 +0200
Subject: [PATCH] CMakeLists.txt: drop -D__STRICT_ANSI__

With gcc-16 and this commit:
https://github.com/gcc-mirror/gcc/commit/ea42e28cbc3d4d579666582e8893e7629bf6a390

libbpnjson fails to build with:
In file included from /OE/build/luneos-blacksail/tmp/work/corei7-64-webos-linux/libpbnjson/2.15.0-22/recipe-sysroot/usr/include/c++/16.2.0/bits/char_traits.h:52,
                 from /OE/build/luneos-blacksail/tmp/work/corei7-64-webos-linux/libpbnjson/2.15.0-22/recipe-sysroot/usr/include/c++/16.2.0/string:45,
                 from /OE/build/luneos-blacksail/tmp/work/corei7-64-webos-linux/libpbnjson/2.15.0-22/sources/libpbnjson-2.15.0-22/include/public/pbnjson/cxx/JErrorHandler.h:20,
                 from /OE/build/luneos-blacksail/tmp/work/corei7-64-webos-linux/libpbnjson/2.15.0-22/sources/libpbnjson-2.15.0-22/src/pbnjson_cxx/JErrorHandler.cpp:17:
/OE/build/luneos-blacksail/tmp/work/corei7-64-webos-linux/libpbnjson/2.15.0-22/recipe-sysroot/usr/include/c++/16.2.0/type_traits:506:12: error: redefinition of 'struct std::__is_integral_helper<__int128>'
  506 |     struct __is_integral_helper<__int128>
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/OE/build/luneos-blacksail/tmp/work/corei7-64-webos-linux/libpbnjson/2.15.0-22/recipe-sysroot/usr/include/c++/16.2.0/type_traits:453:12: note: previous definition of 'struct std::__is_integral_helper<__int128>'
  453 |     struct __is_integral_helper<__GLIBCXX_TYPE_INT_N_0>
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Drop -D__STRICT_ANSI__ to avoid this.

Upstream-Status: Pending [webOS OSE is no longer maintained on github]
Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5edd60a..990dd1c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,7 +32,7 @@ include_directories(include/public)
 
 pkg_check_modules(GLIB2 REQUIRED glib-2.0)
 include_directories(${GLIB2_INCLUDE_DIRS})
-webos_add_compiler_flags(ALL -Wall -D__STRICT_ANSI__ ${GLIB2_CFLAGS_OTHER} -pthread)
+webos_add_compiler_flags(ALL -Wall ${GLIB2_CFLAGS_OTHER} -pthread)
 
 set(WEBOS_USE_WERROR FALSE CACHE BOOL "Set to TRUE to enable -Werror")
 if(WEBOS_USE_WERROR)
